Demo/Demo/ViewModels/PayPalViewModel.swift

```diff
@@ -110,15 +111,30 @@ extension PayPalViewModel: PayPalNativeShippingDelegate {
         withAction shippingActions: PayPalNativeShippingActions
     ) {
         publishStateToMainThread(.mainContent(title: "User action", content: "Shipping address changed", flowComplete: false))
-        shippingActions.approve()
+        if shippingAddress.adminArea1?.isEmpty ?? true || shippingAddress.adminArea1 == "NV" {
+            shippingActions.reject()
+        } else {
+            shippingActions.approve()
+        }
     }
     
     func paypal(
         _ payPalClient: PayPalNativeCheckoutClient,
-        didShippingMethodChange: PayPalNativeShippingMethod,
+        didShippingMethodChange shippingMethod: PayPalNativeShippingMethod,
         withAction shippingActions: PayPalNativeShippingActions
     ) {
         publishStateToMainThread(.mainContent(title: "User action", content: "Shipping method changed", flowComplete: false))
-        shippingActions.approve()
+        Task {
+            do {
+                let shippingMethods = OrderRequestHelpers.getShippingMethods(selectedID: shippingMethod.id)
+                let amount = OrderRequestHelpers.getAmount(shipping: Double(shippingMethod.value ?? "0.0") ?? 0.0)
+                let params = UpdateOrderParams(orderID: orderID, shippingMethods: shippingMethods, amount: amount)
+                try await DemoMerchantAPI.sharedService.updateOrder(params)
+                shippingActions.approve()
+            } catch let error {
+                shippingActions.reject()
```

Comment:
Here is a video of what "reject" looks like if we reject after the shippingMethod change patch order fails.

It's a little odd to me that it reads "Merchant can't ship to this address, choose a different one" when the shipping method was the actual problem. But I think this is the only functionality that NXO exposes right?

https://user-images.githubusercontent.com/35243507/232821844-5f4b353f-d14e-41a4-98fc-bbed66730ae0.mov

Reply:
@scannillo yeap. They dont even offer a why to close the paysheet manually. So this is the only way we can interact with the paysheet if `patch` fails or if we want to display an error

Reply:
Interesting, thanks for checking!